plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AmazonS3Fixture.java

```diff
@@ -46,49 +53,105 @@
  * he implementation is based on official documentation available at https://docs.aws.amazon.com/AmazonS3/latest/API/.
  */
 public class AmazonS3Fixture extends AbstractHttpFixture {
+    private static final String AUTH = "AUTH";
+    private static final String NON_AUTH = "NON_AUTH";
 
     /** List of the buckets stored on this test server **/
     private final Map<String, Bucket> buckets = ConcurrentCollections.newConcurrentMap();
 
     /** Request handlers for the requests made by the S3 client **/
     private final PathTrie<RequestHandler> handlers;
     private final String permanentBucketName;
+    private final String permanentKey;
     private final String temporaryBucketName;
+    private final String temporaryKey;
+    private final String temporarySessionToken;
+    private final String ec2BucketName;
+    private final String ec2ProfileName;
+    private final String ec2Key;
+    private final String ec2Token;
 
     /**
      * Creates a {@link AmazonS3Fixture}
      */
-    private AmazonS3Fixture(final String workingDir, final String permanentBucketName, final String temporaryBucketName) {
+    private AmazonS3Fixture(final String workingDir) {
         super(workingDir);
-        this.permanentBucketName = permanentBucketName;
-        this.temporaryBucketName = temporaryBucketName;
-
-        this.buckets.put(permanentBucketName, new Bucket(permanentBucketName));
-        this.buckets.put(temporaryBucketName, new Bucket(temporaryBucketName));
+        this.permanentBucketName = envVar("S3FIXTURE_PERMANENT_BUCKET_NAME");
+        this.permanentKey = envVar("S3FIXTURE_PERMANENT_KEY");
+        this.temporaryBucketName = envVar("S3FIXTURE_TEMPORARY_BUCKET_NAME");
+        this.temporaryKey = envVar("S3FIXTURE_TEMPORARY_KEY");
+        this.temporarySessionToken = envVar("S3FIXTURE_TEMPORARY_SESSION_TOKEN");
+        this.ec2BucketName = envVar("S3FIXTURE_EC2_BUCKET_NAME");
+        this.ec2ProfileName = envVar("S3FIXTURE_EC2_PROFILE_NAME");
+        this.ec2Key = envVar("S3FIXTURE_EC2_KEY");
+        this.ec2Token = envVar("S3FIXTURE_EC2_TOKEN");
```

Comment:
This _nearly_ feels worthy of abstraction over the various sets of credentials, and I think that'll definitely be worth doing when the ECS-style credentials are added. Optional now, but worth thinking about.

Reply:
agree 💯